Components/DayCalendar.js

```diff
@@ -9,50 +12,251 @@ var {
   ScrollView,
   View,
   Text,
+  PanResponder,
+  Image,
 } = React;
 
+type Event = {
+  title: string;
+  startAt: Date;
+  length: number;
+};
+
 var HOURS_COUNT = 25;
 var HOURS = [];
 for (var i = 0; i < HOURS_COUNT; i++) {
-    var hour = moment().hour(i).second(0).minute(0).format('h:mma');
+    var hour = moment().hour(i).minute(0).second(0).format('h:mma');
     HOURS.push(hour);
 }
+var MINUTES_INA_DAY = 60*24;
+
+var DayCalendar = React.createClass({
+
+  panResponder: {},
+  resizeResponder: {},
+  draggableViewPreviousTop: 0,
+  draggableViewPreviousHeight: 0,
+  draggableView: (null : ?{ setNativeProps(props: Object): void }),
+  resizeHandleView: (null : ?{ setNativeProps(props: Object): void }),
+  draggableViewTopAndHeight: {},
+  scrollView: (null : ?{ setNativeProps(props: Object): void }),
+  totalHeight: 0,
+  beginOfDay: null,
+  endOfDay: null,
+
+  propTypes: {
+    style: View.propTypes.style,
+    eventBoxStyle: View.propTypes.style,
+    day: React.PropTypes.instanceOf(Date).isRequired, 
+    events: React.PropTypes.array.isRequired,
+    newEvent: React.PropTypes.object,
+    timelineHeight: React.PropTypes.number.isRequired,
+    timelineGap: React.PropTypes.number.isRequired,
+    onNewEventStartAtChange: React.PropTypes.func,
+    onNewEventLengthChange: React.PropTypes.func,
+  },
+
+  getDefaultProps: function() {
+    return { day: new Date(), events: [], timelineHeight: 1, timelineGap: 59 };
```

Comment:
Yeah I was wondering about how custom types get passed between the context of the view and the backing functionality. I'd imagine there might be some type of serialization that may take place, and not entirely sure how it would work with `moment` (custom types, as you stated).

No problem :ok_hand: keep it how you have it. Figured I'd throw it out there.

I wonder if React will introduce a "transformer" hook between the binding contexts that let's you perform additional serializing/deserializing steps -- like here, transforming into/from `moment`